### ISSUE TITLE
Edsnlp is interferring with spacy default behavior, and requires "trainable" dependencies

### ISSUE THREAD

**Issue by cvinot** [aphp/edsnlp#234]
## Description

EDSnlp loader from edsnlp 10.0 (AliasLoader in edsnlp __init__.py) is interferring with spacy default loader it seems, which creates additionnal dependencies from the edsnlp trainable components which are not listed, if intended. 

## How to reproduce the bug

**In a fresh environment** 

```poetry add edsnlp ```
download a spacy model, for example [fr_core_news_sm-3.5](https://github.com/explosion/spacy-models/releases/tag/fr_core_news_sm-3.5.0
)

then:
```python
from spacy import load
model = "fr_core_news_sm-3.5"
nlp = load(model)
```
Here it raises errors depending on the environment, "foldedtensor" or in our case "torch" module not found. which is not a dependency in any package, as edsnlp[ml] was not desired.

## Other
I found the bug with this snippet:
```python
import spacy
model = "fr_core_news_sm-3.5"
nlp = spacy.load(model)
```
Which raises ```[E002] Can't find factory for 'tok2vec' for language French (fr).``` and shows no relation to edsnlp whatsoever interestingly enough.

**Comment by percevalw** [comment]
Hi, thank you for letting us know about this ! This is indeed problematic. By default, spacy imports all available components (i.e., that have a `spacy_factories` entrypoint namespace), even though they are not required by the user. You did not install `edsnlp[ml]` indeed, so importing some components like `eds.transformer` will fail, hence the bug you reported.

You can try with to re-install edsnlp with the fix from #235, let me know if this works for you !

```
pip install git+https://github.com/aphp/edsnlp@fix-non-ml-install
```

**Comment by cvinot** [comment]
Interesting, I never dived into these depths of spacy. 

It does fix the issue for me at first glance. Great reactivity, thanks.

**Comment by cvinot** [comment]
I still have the same problem if I import part of edsnlp though. 

```
import spacy
import edsnlp
model = "fr_core_news_sm-3.5"
nlp = spacy.load(model)
```

raises the same error.

**Comment by percevalw** [comment]
Ok thank you for testing, the new fix in #235 (same `pip install git+https://github.com/aphp/edsnlp@fix-non-ml-install`) should do the trick now 🤞
